arm-linux FTBFS in CI

**Environment**
- libunwind version: master
- architecture: arm, armhf
- OS: linux-gnu

**Description**
*arm-linux-gnueabi* and *arm-linux-gnueabihf* targets are failing to build in CI on Ubuntu 22.04 with GCC 12 (and manually on Ubuntu 20.04 with GCC 10). The following errors occur.
```
Making all in src
make[1]: Entering directory '/home/runner/work/libunwind/libunwind/src'
make  all-am
make[2]: Entering directory '/home/runner/work/libunwind/libunwind/src'
  CC       os-linux.lo
  CC       dl-iterate-phdr.lo
  CC       elf32.lo
  CC       arm/Gos-linux.lo
  CC       mi/Gaddress_validator.lo
  CC       mi/Gdestroy_addr_space.lo
  CC       mi/Gdyn-extract.lo
  CC       mi/Gdyn-remote.lo
  CC       mi/Gfind_dynamic_proc_info.lo
mi/Gaddress_validator.c: In function ‘_do_pipe2’:
mi/Gaddress_validator.c:4[8](https://github.com/libunwind/libunwind/actions/runs/8961845312/job/24629192202?pr=753#step:6:9):10: warning: implicit declaration of function ‘pipe2’; did you mean ‘pipe’? [-Wimplicit-function-declaration]
   48 |   return pipe2 (pipefd, O_CLOEXEC | O_NONBLOCK);
      |          ^~~~~
      |          pipe
  CC       mi/Gget_accessors.lo
  CC       mi/Gget_fpreg.lo
  CC       mi/Gget_proc_info_by_ip.lo
  CC       mi/Gget_proc_name.lo
  CC       mi/Gget_reg.lo
  CC       mi/Gput_dynamic_unwind_info.lo
  CC       mi/Gset_cache_size.lo
  CC       mi/Gset_caching_policy.lo
  CC       mi/Gset_fpreg.lo
  CC       mi/Gset_iterate_phdr_function.lo
  CC       mi/Gset_reg.lo
  CC       mi/Gget_elf_filename.lo
  CC       arm/Gapply_reg_state.lo
  CC       arm/Gcreate_addr_space.lo
mi/Gset_iterate_phdr_function.c: In function ‘_Uarm_set_iterate_phdr_function’:
mi/Gset_iterate_phdr_function.c:18:33: error: ‘dl_iterate_phdr’ undeclared (first use in this function)
   18 |     as->iterate_phdr_function = dl_iterate_phdr;
      |                                 ^~~~~~~~~~~~~~~
mi/Gset_iterate_phdr_function.c:18:33: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [Makefile:4803: mi/Gset_iterate_phdr_function.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/runner/work/libunwind/libunwind/src'
make[1]: *** [Makefile:2[9](https://github.com/libunwind/libunwind/actions/runs/8961845312/job/24629192202?pr=753#step:6:10)21: all] Error 2
make[1]: Leaving directory '/home/runner/work/libunwind/libunwind/src'
make: *** [Makefile:622: all-recursive] Error 1
Error: Process completed with exit code 2.
```

**Reproduction Steps**
Steps to reproduce the behavior:
1. Check out mainline
2. Run `autoreconf -if`
3. Run `./configure --prefix=/usr --build=x86_64-linux-gnu --host=arm-linux-gnueabihf`
4. See error

--- a/configure.ac
+++ b/configure.ac
@@ -65,8 +65,8 @@ AC_CHECK_HEADERS(asm/ptrace_offsets.h asm/ptrace.h asm/vsyscall.h endian.h sys/e
 
 dnl Set target-specific compile flags
 AS_CASE([$target_os],
-        [*-gnu],    [UNW_TARGET_CPPFLAGS="-D_GNU_SOURCE"],
-        [solaris*], [UNW_TARGET_CPPFLAGS="-D__EXTENSIONS__"]
+        [*-gnu|*-gnueabi*], [UNW_TARGET_CPPFLAGS="-D_GNU_SOURCE"],
+        [solaris*],         [UNW_TARGET_CPPFLAGS="-D__EXTENSIONS__"]
 )
 AC_SUBST([UNW_TARGET_CPPFLAGS])
 

